library/src/image_c_api.c

```diff
@@ -0,0 +1,166 @@
+#include "image_c_api.h"
+#include "image.h"
+#include <cstring>
+
+struct f3d_image {
+    f3d::image img;
+};
+
+f3d_image_t* f3d_image_new(void) {
```

Comment:
I think this file (which includes and uses `image.h`) must be a C++ file. Only `image_c_api.h` have to be C only (and the unit test of course).